plugins/modules/ipbase_info.py

```diff
@@ -0,0 +1,163 @@
+  - "Retrieve IP geolocation and other facts of a host's IP address using the ipbase.com API"
+author: "Dominik Kukacka (@dominikkukacka)"
+extends_documentation_fragment:
+  - community.general.attributes
+  - community.general.attributes.info_module
+options:
+  ip:
+    description:
+      - "The IP you want to get the info for."
+    required: false
+    type: str
+  apikey:
+    description:
+      - "The apikey for the request if you need more requests."
+    required: false
+    type: str
+notes:
+  - "Check U(https://ipbase.com/) for more information."
+'''
+
+EXAMPLES = '''
+- name: "Get IP geolocation information of the primary outgoing IP"
+  community.general.ipbase_info:
+  register: my_ip_info
+
+- name: "Get IP geolocation information of a specific IP"
+  community.general.ipbase_info:
+    ip: "8.8.8.8"
+  register: my_ip_info
+'''
+
+RETURN = '''
+data:
+    description: "The data retrieved from ipbase.com."
+    returned: success
+    type: dict
+    sample: {
+        "as_name": "T-Mobile Austria GmbH",
+        "as_number": 8412,
+        "city": "Vienna",
+        "continent": "Europe",
+        "continent_code": "EU",
+        "country": "Austria",
+        "country_code": "AT",
+        "hostname": "81-223-100.100.static.upcbusiness.at",
+        "ip": "81.223.100.100",
+        "is_in_european_union": true,
+        "latitude": 48.1861686706543,
+        "longitude": 16.403240203857422,
+        "region": "Vienna",
+        "region_code": "AT-9",
+        "timezone": "Europe/Vienna",
+        "zip": "1030"
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.urls import fetch_url
+
+
+USER_AGENT = 'ansible-ipbase-module/0.1.0'
+BASE_URL = 'https://api.ipbase.com/v2/info?hostname=1'
+
+
+class IpbaseInfo(object):
+
+    def __init__(self, module):
+        self.module = module
+
+    def _get_url_data(self, url):
+        response, info = fetch_url(
+            self.module,
+            url,
+            force=True,
+            timeout=10,
+            headers={
+                'Accept': 'application/json',
+                'User-Agent': USER_AGENT,
+            })
+
+        if (info['status'] != 200):
+            self.module.fail_json(msg='The API request to ipbase.com returned an error status code {0}'.format(info['status']))
+        else:
+            try:
+                content = response.read()
+                result = self.module.from_json(content.decode('utf8'))
+            except ValueError:
+                self.module.fail_json(
+                    msg='Failed to parse the ipbase.com response: '
+                    '{0} {1}'.format(url, content))
+            else:
+                return result
+
+    def info(self):
+
+        ip = self.module.params['ip']
+        apikey = self.module.params['apikey']
+
+        url = BASE_URL
+        if ip:
+            url += '&ip=' + str(ip)
+
+        if apikey:
+            url += '&apikey=' + str(apikey)
```

Comment:
Instead of manually composing the URL (without any escaping), you should use standard Python functions that do exactly that:
```
params = {}
if ip:
    params['ip'] = ip
...
if params:
    url += '&' + urlencode(params)
```
For that you need `from ansible.module_utils.six.moves.urllib.parse import urlencode`.